microsoft/testsuites/dpdk/dpdksuite.py

```diff
@@ -817,6 +829,8 @@ def initialize_node_resources(
     pmd: str,
     sample_apps: Union[List[str], None] = None,
 ) -> DpdkTestResources:
+
+    validate_node_requirements(node)
```

Comment:
It can be done by configuring in requirement of metadata. It can help filter vm sizes before environment is deployed, so it's much faster, and can find testable vm sizes automatically if it's not specified.